### PR TITLE
Dinathom/PSO Operator

### PR DESCRIPTION
**Pull request by dinathom**
New PSO operator install method for OpenShift 3.11+ cluster

This PR creates a helm-chart based operator for pure-k8s-plugin helm-chart.

### CODE REVIEW

File: operator/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+**OpenShift Note**: In OpenShift 3.11, the default node-selector for a project does not allow PSO-operator to mount volumes on master and infra nodes. 
+If you want to mount volumes on master and infra nodes OR run pods in the default namespace using PSO-volumes, then modify the install script as follows.<br\>
+```
+Existing line:     $KUBECTL adm new-project ${NAMESPACE}
+
+Change to allow volume mounts on master and infra nodes:     $KUBECTL adm new-project ${NAMESPACE} --node-selector=""
+```
+
+2. Create a Custom Resource Definition (CRD) for the PSO Operator. <br/>
+The script waits for the CRD to be published in the cluster. If adter 10 seconds the API server has not setup the CRD, the script times out. To wait longer, pass the parameter
```
Comment (trevormeiss) [inline operator/README.md]: [typo] "after"

File: operator/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+In addition, the operator needs access to multiple resources in the project/namespace that it is deployed in to function correctly. Hence it is recommended to install the PSO-operator in the non-default namespace.
+<br/>
+<br/>
+   
+4. Creates a deployment for the Operator.<br/>
+Finally the script creates and deploys the operator using the customized parameters passed in the ``values.yaml`` file.
+
+
+## Uninstall
+To uninstall the PSO Operator, delete the project/namespace in which the PSO operator is installed. It is **strongly recommended** to install the PSO Operator in a new project and not add any other pods to this project/namespace. Any pods in this project will be cleaned up on an uninstall.
```
Comment (trevormeiss) [inline operator/README.md]: Should we include a sample command?

File: operator/install.sh [hunk 3]
```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env bash
+IMAGE=quay.io/purestorage/pso-operator:v0.0.1
+NAMESPACE=psoOperator
```
Comment (trevormeiss) [inline operator/install.sh]: Isn't `kebab-case` more common for namespaces?

File: operator/install.sh [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            - name: OPERATOR_NAME
+              value: "pso-operator"
+' | sed "s|REPLACE_IMAGE|${IMAGE}|" | ${KUBECTL_NS} -
+
+# 5. Use the values.yaml file to create a customized PSO operator instance
+( echo '
+apiVersion: purestorage.com/v1
+kind: PSOPlugin
+metadata:
+  name: psoplugin-operator
```
Comment (trevormeiss) [inline operator/install.sh]: Do we need a namespace here?

File: operator/install.sh (hunk 3 above)
Comment (sdodsley) [reply operator/install.sh]: Tend to agree with this comment.

**Comment by dinathom** [comment]
> I would make sure that when this is published the legacy helm stuff and docker commands in the test section are not included.

@sdodsley this repo will have both the helm-chart and the Operator based on the helm-chart. Hence a new directory for the operator. I will add this to the README.

**Comment by sdodsley** [comment]
LGTM

**Pull request merged**